recipes/aws-libfabric/all/conanfile.py

```diff
@@ -0,0 +1,99 @@
+from conans import ConanFile, AutoToolsBuildEnvironment, tools
+from conans.errors import ConanInvalidConfiguration
+import os
+
+required_conan_version = ">=1.35.0"
+
+class LibfabricConan(ConanFile):
+    name = "aws-libfabric"
+    description = "AWS Libfabric"
+    topics = ("fabric", "communication", "framework", "service")
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/aws/libfabric"
+    license = "BSD-2-Clause", "GPL-2.0-or-later"
+    settings = "os", "arch", "compiler", "build_type"
+    _providers = ['gni', 'psm', 'psm2', 'psm3', 'rxm', 'sockets', 'tcp', 'udp', 'usnic', 'verbs', 'bgq']
+    options = {
+        **{ p: "ANY" for p in _providers },
+        **{
+            "shared": [True, False],
+            "fPIC": [True, False],
+            "with_libnl": "ANY",
+            "with_bgq_progress": [None, "auto", "manual"],
+            "with_bgq_mr": [None, "basic", "scalable"]
+        }
+    }
+    default_options = {
+        **{ p: "auto" for p in _providers },
+        **{
+            "shared": False,
+            "fPIC": True,
+            "with_libnl": None,
+            "with_bgq_progress": None,
+            "with_bgq_mr": None
+        }
+    }
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    _autotools = None
+
+    def config_options(self):
+        if self.settings.os == 'Windows':
+            del self.options.fPIC
+
+    def configure(self):
+        if self.options.shared:
+            del self.options.fPIC
+        del self.settings.compiler.libcxx
+        del self.settings.compiler.cppstd
+
+    def validate(self):
+        if self.settings.compiler == "Visual Studio":
+            raise ConanInvalidConfiguration("The libfabric package cannot be built on Visual Studio.")
+        for p in self._providers:
+            if self.options.get_safe(p) not in ["auto", "yes", "no", "dl"] and not os.path.isdir(str(self.options.get_safe(p))):
```

Comment:
Is this meaning of the option values correct?
- `auto`: detect for presence of the dependency and enable if present, disable if not present
- `yes`: enable it
- `no`: disable it
- `dl`: download the dependency, and enable it

If it is correct, then the options are not following cci idiology.
cci does not allow downloading sources in other places then in the `source` method.

It's important to have all dependencies available as a system package.
Like libnl, if all/some dependencies become cci packages, then these options can become boolean values.

Reply:
the sources for the dependencies are part of the package, e.g. https://github.com/aws/libfabric/tree/master/prov/bgq/src

Reply:
Can't they just be made boolean and passed `yes` or `no`?

Reply:
`dl` means build provider as a shared library.

Reply:
Ah yes. Does it need `dl` as a system library added in `package_info` in that case?